compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -627,7 +627,8 @@ def _do_start_endpoint(
         )
 
     reg_info = {}
-    config_str = None
+    config_str: str | None = None
+    fn_allow_list: list[str] | None | int = 0
```

Comment:
I'll have to think about the implications of the need to use '0' instead of using None.  That's a special case consideration that is kinda hidden.  It seems like it's ok for this use case, but I feel like some documentation, even in code (maybe on the manager side where it's passed in?) would keep its meaning better known.

Reply:
Ah, no, no documentation.  The `0` is an internal representation detail strictly for this function.  Problem: what is the difference between the following two stdin documents, from the perspective of `fn_allow_list`:
```json
{"allowed_functions": null}
```

```json
{}
```

The canary's purpose is to explicitly allow `null`/`None` as an option via `stdin` (i.e., "no disallowed functions"), which is different from `allowed_functions = []` (the empty list).

Reply:
..

And we've now talked.  Simply naming the canary as opposed to using a magic number FTW.